utils/stall_free.hh

```diff
@@ -56,14 +58,99 @@ void merge_to_gently(std::list<T>& list1, const std::list<T>& list2, Compare com
     }
 }
 
-template<class T>
-seastar::future<> clear_gently(std::list<T>& list) {
-    return repeat([&list] () mutable {
-        if (list.empty()) {
-            return seastar::stop_iteration::yes;
-        }
-        list.pop_back();
-        return seastar::stop_iteration::no;
+template <typename T>
+concept GentlyClearable = requires (T x) {
+    { x.clear_gently() } -> std::same_as<future<>>;
+};
+
+template <typename Ptr, typename T = typename Ptr::element_type>
+concept SmartPointer = requires (Ptr x) {
+    { x.get() } -> std::same_as<T*>;
+};
+
+template <typename Vector, typename T = typename Vector::value_type>
+concept VectorClass = requires (Vector x) {
```

Comment:
It's vector-like, not necessarily a vector. std::dequeue fits and isn't a vector. 

No need for T, you can have a typename requirement below.

Reply:
Fixes in v5